index.ts

```diff
@@ -1,35 +1,60 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private _keys: string[] | undefined
+  private _original: object[] = []
+  private _transposed: { [key: string]: any } = {}
+
+  constructor(data?: object[], options?: { keys?: string[] }) {
+    if (data !== undefined) { this.push(...data) }
+    if (options !== undefined) {
+      if (options.keys !== undefined) { this._keys = options.keys }

```

Comment:
Just realised I can safely remove this undefined check (`options.keys !== undefined`)

Reply:
Go for it :100:

Reply:
just realised, you have an `if` in an `if`, and neither are needed:

```ts
if (options !== undefined) {
  if (options.keys !== undefined) { this._keys = options.keys }
}
```

change this to
```ts
this._keys = options && options.keys || []
```

Reply:
Man I love JS.
Will change it to 
`this._keys = options.keys || []`
as we'd be duplicating `options &&` when we add more options

Reply:
I'm going to use 
`if (options !== undefined) {
  this._keys = options.keys
}`
Then we won't have to do `options &&` for each new option we have,  and I actually rely on `_keys` being undefined rather than empty for `(this._keys || Object.keys(datum))` in `transpose`.

Reply:
Then stop relying on `_keys` being `undefined` - Was this a problem with the original `transpose` implementation?

Reply:
Yes. The new transpose is fundamentally different: I'm adding keys based on the object unless `_keys` is defined, in which case I'm only transposing those.

Reply:
So it sounds like the only difference is how it selects keys?

Reply:
Whoops, missed this. Yes, which is why the new version looked had more `if`s than before. 
Outdated.